code/game/machinery/camera/tracking.dm

```diff
@@ -68,57 +68,45 @@
 		trackable_mobs()
 
 	var/datum/weakref/target = (isnull(track.humans[target_name]) ? track.others[target_name] : track.humans[target_name])
+	ai_start_tracking(target.resolve())
 
-	ai_actual_track(target.resolve())
-
-/mob/living/silicon/ai/proc/ai_actual_track(mob/living/target)
-	if(!istype(target))
+/mob/living/silicon/ai/proc/ai_start_tracking(mob/living/target) //starts ai tracking
+	if(!target || !target.can_track(src))
+		to_chat(src, "<span class='warning'>Target is not near any active cameras.</span>")
 		return
-	var/mob/living/silicon/ai/U = usr
-
-	U.cameraFollow = target
-	U.tracking = 1
-
-	if(!target || !target.can_track(usr))
-		to_chat(U, "<span class='warning'>Target is not near any active cameras.</span>")
-		U.cameraFollow = null
+	if(ai_tracking_target) //if there is already a tracking going when this gets called makes sure the old tracking gets stopped before we register the new signals
+		ai_stop_tracking()
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/ai_stop_tracking)
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, .proc/ai_actual_track)
+	ai_tracking_target = target
+	eyeobj.setLoc(get_turf(target)) //on the first call of this we obviously need to jump to the target ourselfs else we would go there only after they moved once
+	to_chat(src, "<span class='notice'>Now tracking [target.get_visible_name()] on camera.</span>")
+
+/mob/living/silicon/ai/proc/ai_stop_tracking(var/reacquire_failed = FALSE) //stops ai tracking
```

Comment:
Since this is being used as a signal handler coming in from RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/ai_stop_tracking)
the parameter 'reacquire_failed' will actually be datum/source when called from the mob being deleted. This will probably still work, but it should use a wrapper with the correct parameters for COMSIG_PARENT_QDELETING to prevent any weirdness and to make the code cleaner

Reply:
True i totally forgot about that part of signal behavior not hard to fix trough

Reply:
yeah i only really need to fix the parameter offset since the only time reaquire_failed can and should be true is if it gets called by the CALLBACK timer

Reply:
Turns out the COMSIG_PARENT_QDELETING can also send `SEND_SIGNAL(D, COMSIG_PARENT_QDELETING, force) // Let the (remaining) components know about the result of Destroy` the force argument so it needs to that also needs to be taken into account.

Reply:
I could also wrap the signal proc but that would still require the same offsets here